### PR TITLE
CORE-8622 Add `TransactionMetadata.getNumberOfComponentGroups`

### PR DESCRIPTION
**Pull request by lankydan**

### CODE REVIEW

File: ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionMetadata.kt [hunk 1]
```diff
@@ -1,57 +1,59 @@
...
+     * Gets the digest settings used to calculate the transaction hashes
      *
-     * @return digest settings as map
+     * @return The digest settings as map
      */
     fun getDigestSettings(): LinkedHashMap<String, Any>
 
     /**
-     * Version of the metadata JSON schema to parse this metadata entity
+     * Gets of the metadata JSON schema to parse this metadata entity
```
Comment (vlajos) [inline ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionMetadata.kt]: This does not look right.

Comment (lankydan) [reply ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionMetadata.kt]: ...

File: ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionMetadata.kt [hunk 2]
```diff
@@ -1,57 +1,59 @@
...
-     * Get information about the contract CPKs governing the transaction (installed on the
+     * Gets information about the contract CPKs governing the transaction (installed on the
      * virtual node when the transaction was created)
      *
-     * @return list of CPK summaries
+     * @return A list of CPK summaries
      */
     fun getCpkMetadata(): List<CordaPackageSummary>
 
+    fun getComponentGroupIndexes(): List<Int>
```
Comment (vlajos) [inline ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionMetadata.kt]: I suppose this would need a doc line.

**Review by vlajos — DISMISSED** [review]

**Review by blsemo — DISMISSED** [review]
LGTM

**Review by vlajos — APPROVED** [review]

**Review by blsemo — APPROVED** [review]

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**